### ISSUE TITLE
milkdrop presets using textures/images doesn't loading

### ISSUE THREAD

**Issue by joenio** [projectM-visualizer/projectm#310]
Hello guys,

I tried some milkdrop presets using textures with images and none of the tests I did works, projectM always return error like below:

```
Syntax error: expected ';' near '.'
Failed to parse HLSL(step2) Warp shader
```

Anyone have good experience with presets using textures in projectM? projectM support this kind of milkdrop preset?

Thanks in advance!

**Comment by auralarch** [comment]
I don't believe this feature/function has ever worked properly.  At least not that I know of, not in the Open Source branch anyway (the guys who started this project never shared the updates that they wrote which got their release for mobile devices working well enough to seduce users to buy their paid version).

Although I am by NO means a voice of any authority on this project.  But since nobody had responded in almost a month I figured I'd share my experience with you.

**Comment by revmischa** [comment]
I have seen images (like a penguin) appear in some presets, and I assume that the noise samplers must be working? But I have no special knowledge here. It's also possible that some of the texture sampling code was broken or made more broken by the conversion for immediate mode to VBO and GLES compatibility. 
Hopefully someone who knows something about shaders and textures and openGL can take a look at the existing code and the validity of the texture mapping code output by the HLSL->GLSL transpiler.

**Comment by caban** [comment]
Textures / images do work. The current build just doesn't provide any images. You need to copy a folder named "textures" to the same folder as projectm.exe.

I am using a combination of the "textures" folder in the 52k preset pack and I also extracted images from the projectM Android SDK.

I've also started preparing a list of missing images in the 52k preset pack.

**Comment by milkdropper** [comment]
(I posted with the wrong account before... oops)

Textures / images do work. The current build just doesn't provide any images. You need to copy a folder named "textures" to the same folder as projectm.exe.

I am using a combination of the "textures" folder in the 52k preset pack and I also extracted images from the projectM Android SDK.

I've also started preparing a list of missing images in the 52k preset pack.

**Comment by auralarch** [comment]
I love being wrong!  That is fantastic…

It is delightful and exciting to see how this project is progressing!

Perhaps some presets which implement textures should be included in the 'default' collection which is automatically installed so that the directory will be created in the proper spot with the correct name and stocked with the images called for by presets in the collection?

**Comment by kblaschke** [comment]
One thing is that a few presets shipped with projectM reference textures that aren't part of the preset directories.

The other, larger issue is that projectM doesn't provide th built-in noise textures and has no support for random texture sampling.

See the new issue #532 for more details and potential progress on that matter.
